python/social_bot/gazebo_agent.py

```diff
@@ -180,10 +195,37 @@ def get_full_states_observation(self, teacher):
         """
         task_specific_ob = teacher.get_task_specific_observation(self)
         agent_pose = np.array(self.get_pose()).flatten()
-        agent_vel = np.array(self.get_velocities()).flatten()
-        internal_states = self.get_internal_states()
-        obs = np.concatenate(
-            (task_specific_ob, agent_pose, agent_vel, internal_states), axis=0)
+        if self._use_simple_full_states:
+            # assumes GoalTask and that the first 3 dims of the
+            # task_specific_observation give the goal position.
+            yaw = agent_pose[5]
+            obs = []
+            while len(task_specific_ob) > 1:
+                x = task_specific_ob[0] - agent_pose[0]
+                y = task_specific_ob[1] - agent_pose[1]
+                # rotate -yaw
+                rotated_x = x * np.cos(-yaw) - y * np.sin(-yaw)
+                rotated_y = x * np.sin(-yaw) + y * np.cos(-yaw)
+                if self._view_angle_limit > 0.001:
+                    dist = math.sqrt(rotated_x * rotated_x + rotated_y * rotated_y)
+                    rotated_x /= dist
+                    rotated_y /= dist
+                    magnitude = 1. / dist
```

Comment:
Why not use dist directly here? If agent get very closer to the goal, dist is very small, will magnitude be very large?

Reply:
Good point.  Dist is usually not very close to 0 due to success dist thresh.  The reason to use magnitude is to make it change monotonically when object goes out of view, in which case the magnitude becomes 0.  I couldn't find a natural value if we use dist.